OpenOversight/app/main/views.py

```diff
@@ -866,173 +867,69 @@ def edit_department(department_id: int):
-
-    unit_selections = ["Not Sure"] + [
-        uc[0]
-        for uc in db.session.query(Unit.description)
-        .filter_by(department_id=department_id)
+    form.unit.query = list(
+        Unit.query.filter_by(department_id=department_id)
         .order_by(Unit.description.asc())
         .all()
-    ]
-    rank_selections = [
-        jc[0]
-        for jc in db.session.query(Job.job_title, Job.order)
-        .filter_by(department_id=department_id)
-        .order_by(Job.job_title)
-        .all()
-    ]
-    if (units := request.args.getlist("unit")) and all(
-        unit in unit_selections for unit in units
-    ):
-        form_data["unit"] = units
-    if (ranks := request.args.getlist("rank")) and all(
-        rank in rank_selections for rank in ranks
-    ):
-        form_data["rank"] = ranks
-    if current_job_arg := request.args.get("current_job"):
-        form_data["current_job"] = current_job_arg
+    ) + [Unit(id=None, description="Not Sure")]
+    form.process(request.args)
+    form_data = form.data
 
-    officers = filter_by_form(form_data, Officer.query, department_id).filter(
-        Officer.department_id == department_id
-    )
+    officers = filter_by_form(form_data, Officer.query, department_id)
 
     # Filter officers by presence of a photo
     if form_data["require_photo"]:
         officers = officers.join(Face)
     else:
         officers = officers.options(selectinload(Officer.face))
 
+    officers = officers.options(selectinload(Officer.incidents))
+
     officers = officers.order_by(Officer.last_name, Officer.first_name, Officer.id)
 
     officers = officers.paginate(
-        page=page, per_page=current_app.config[KEY_OFFICERS_PER_PAGE], error_out=False
+        page=form_data["page"],
+        per_page=current_app.config[KEY_OFFICERS_PER_PAGE],
+        error_out=False,
     )
 
+    img_id_to_officer = {}
     for officer in officers.items:
-        officer_face = sorted(officer.face, key=lambda x: x.featured, reverse=True)
-
-        # Could do some extra work to not lazy load images but load them all together.
-        # To do that properly we would want to ensure to only load the first picture of
-        # each officer.
-        if officer_face and officer_face[0].image:
-            officer.image = officer_face[0].image.filepath
-
-    choices = {
-        "race": RACE_CHOICES,
-        "gender": GENDER_CHOICES,
-        "rank": [(rc, rc) for rc in rank_selections],
-        "unit": [(uc, uc) for uc in unit_selections],
-    }
-
-    next_url = url_for(
+        if officer.face:
+            face_image_id = sorted(
+                officer.face, key=lambda x: x.featured, reverse=True
+            )[0].img_id
+            img_id_to_officer[face_image_id] = officer
+    images = Image.query.filter(Image.id.in_(img_id_to_officer.keys())).all()
+    for image in images:
+        img_id_to_officer[image.id].image = serve_image(image.filepath)
```

Comment:
On my end, I'm seeing that many officers who have images available are showing up as "No Photos" in the browse view. This is happening because multiple officers have the same image assigned to them.

I'm not sure that this happens in practice since we usually use face thumbnails but it might be an edge case want to account for.

Reply:
Yeah, that shouldn't happen in practice, but I agree with the concern and will make adjustments

Reply:
Fixed this, based on my tests the issue should be resolved